src/main/scala/org/bitcoins/core/gen/ScriptGenerators.scala

```diff
@@ -73,36 +103,64 @@ trait ScriptGenerators extends BitcoinSLogger {
     multiSignatureScriptPubKey
   }
 
+  //TODO: Improve the format of each scriptPubKey generator to return a privateKey needed to spend it.
   def p2shScriptPubKey : Gen[P2SHScriptPubKey] = for {
     randomScriptPubKey <- pickRandomNonP2SHScriptPubKey
   } yield P2SHScriptPubKey(randomScriptPubKey)
 
-
   def emptyScriptPubKey = p2pkScriptPubKey.map(_ => EmptyScriptPubKey)
 
   def pickRandomNonP2SHScriptPubKey : Gen[ScriptPubKey] = {
-    val randomNum = (scala.util.Random.nextInt() % 3).abs
+    val randomNum = (scala.util.Random.nextInt() % 5).abs
     if (randomNum == 0) p2pkScriptPubKey
     else if (randomNum == 1) p2pkhScriptPubKey
+    else if (randomNum == 2) cltvScriptPubKey
+    else if (randomNum == 3) csvScriptPubKey
     else multiSigScriptPubKey
   }
 
+  def pickRandomNonLockTimeScriptPubKey(pubKey : ECPublicKey) : Gen[ScriptPubKey] = {
+    val randomNum = (scala.util.Random.nextInt() % 3).abs
+    if (randomNum == 0) P2PKScriptPubKey(pubKey)
+    else if (randomNum == 1) P2PKHScriptPubKey(pubKey)
+    else P2PKHScriptPubKey(pubKey)
+  }
+
+  def pickRandomNonLockTimeScriptPubKey : Gen[(ScriptPubKey, Seq[ECPrivateKey])] = {
+    val num = (scala.util.Random.nextInt() % 5).abs
+    val isMulti : Boolean = if (num == 4) true else false
+      if (isMulti)  {
```

Comment:
As of latest commit (https://github.com/bitcoin-s/bitcoin-s-core/pull/31/commits/ea2229946e92b46ba815f5ab8485a1e4bda39b82), the respective scriptPubKey generators don't return the ECPrivateKey(s) needed to satisfy the script, rendering the functionality useless for generating signedScriptSignatures. Currently working on this.